### PR TITLE
add shape check for rrelu_with_noise

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122870

Fix https://github.com/pytorch/pytorch/issues/121094.
Add shape check for rrelu_with_noise, check whether the shape of input tensor and noise tensor are the same.

### CODE REVIEW

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -606,6 +606,7 @@ Tensor& rrelu_with_noise_out_cpu(const Tensor& self,
     bool training,
     c10::optional<Generator> generator,
     Tensor& output) {
+  TORCH_CHECK(self.sizes() == noise.sizes(), "noise tensor shape must match self tensor shape. Got self.shape = ", self.sizes(), " noise.shape = ", noise.sizes())
```
Comment (Skylion007) [inline aten/src/ATen/native/Activation.cpp:609]: Should this be `sym_sizes()` to prevent issues with symbolic tensors?

Comment (jiayisunx) [reply aten/src/ATen/native/Activation.cpp]: Fixed, thanks!

**Comment by jiayisunx** [comment]
@Skylion007, could you please review this PR again? Thanks.

File: aten/src/ATen/native/Activation.cpp [hunk 2]
```diff
@@ -606,6 +606,7 @@ Tensor& rrelu_with_noise_out_cpu(const Tensor& self,
     bool training,
     c10::optional<Generator> generator,
     Tensor& output) {
+  TORCH_CHECK(self.sym_sizes() == noise.sym_sizes(), "noise tensor shape must match self tensor shape. Got self.shape = ", self.sym_sizes(), " noise.shape = ", noise.sym_sizes())
```
Comment (mingfeima) [inline aten/src/ATen/native/Activation.cpp:609]: TORCH_CHECK does not need a `;` to be compiled correctly, but usually we still add a `;` at the end

**Review by mingfeima — APPROVED** [review]

**Comment by jiayisunx** [comment]
@cpuhrsch  , could you please review this PR? Thanks.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**